common/src/main/java/com/wynntils/wc/utils/SpellManager.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wc.utils;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.mc.event.ClientTickEvent;
+import com.wynntils.mc.event.TitleSetTextEvent;
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class SpellManager {
+    private static final Pattern SPELL_PATTERN = Pattern.compile(
+            "§([LR]|Right|Left)§-§([LR?]|Right|Left)§-§([LR?]|Right|Left)§".replace("§", "(?:§[0-9a-fklmnor])*"));
+    public static final boolean[] NO_SPELL = new boolean[0];
```

Comment:
I realize this is how it was done in Legacy, but I'm not really fond of it. The spells are not properly "boolean". Yes, I understand it's a two-state variable, but they should be enums, since they carry no meaningful truth value.

Reply:
I went with the name SpellUnit.